platforms/documentation/docs/src/docs/userguide/authoring-builds/tasks/lazy_configuration.adoc

```diff
@@ -321,6 +321,58 @@ $ gradle show
 include::{snippetsPath}/providers/propertyConvention/tests/propertyConvention.out[]
 ----
 
+=== Where to apply conventions from?
+
+There are several appropriate locations for setting a convention on a property at configuration time (i.e., before execution).
+
+====
+include::sample[dir="snippets/providers/propertyConventionCallsites/kotlin",files="build.gradle.kts[tags=convention-callsites]"]
+include::sample[dir="snippets/providers/propertyConventionCallsites/groovy",files="build.gradle[tags=convention-callsites]"]
+====
+
+==== From a plugin's `apply()` method
+
+Plugin authors may configure a convention on a lazy property from a plugin's `apply()` method, while performing preliminary configuration of the task or extension defining the property.
+This works well for regular plugins (meant to be distributed and used in the wild), and internal <<sharing_build_logic_between_subprojects.adoc#sec:convention_plugins,convention plugins>> (which often configure properties defined by third party plugins in a uniform way for the entire build).
+
+====
+include::sample[dir="snippets/providers/propertyConventionCallsites/kotlin",files="build.gradle.kts[tags=convention-callsites-from-plugin]"]
+include::sample[dir="snippets/providers/propertyConventionCallsites/groovy",files="build.gradle[tags=convention-callsites-from-plugin]"]
+====
+
+==== From a build script
```

Comment:
🤔 Maybe instead of "From a build script" rather From a convention plugins"?

Convention plugins are normally used for sharing build logic. Build script sounds very broad term.

Reply:
@asodja Convention plugins are (now) covered in the previous section.

The different sections attempt to cover alternatives across syntax/API options and less about types of users.